content/projects/gsoc/2018/remoting-over-message-bus.adoc

```diff
@@ -26,57 +26,121 @@ links:
-* Decide whether we use Kafka in the final implementation (or not).
-* Create a blogpost with a PoC introduction.
-
-Second phase:
-
-* Support security for master-agent connection:
-** Kafka authentication/authorization.
-** Agent secrets.
-* Improve Kafka producer-consumer model to ensure reliability.
-* Bug fixing.
-* Release alpha version and address feedback.
+This project aims to develop a plugin in order to add support of a popular message queue/bus technology (Kafka) as a fault-tolerant communication layer in Jenkins.
+
+The plugin source code can be found in https://github.com/jenkinsci/remoting-kafka-plugin[GitHub].
+
+=== Benefits to the community
+The plugin provide useful features to the community:
+
+* Provide a new method to connect agent to master using Kafka besides existing methods such as JNLP or ssh-slaves-plugin.
+* Help to resolve the existing issues with the TCP protocol between master and agent communication in Jenkins.
+* Help to resolve traffic prioritization and multi-agent communications issue in Jenkins.
+
+=== Why Kafka?
+* Kafka itself is not a queue like ActiveMQ or RabbitMQ, it is a distributed, replicated commit log. This helps to remove message delivery complexity we have in traditional queue system.
+* We need to support data streaming as a requirement, and Kafka is good at this aspect, which RabbitMQ is lack of.
+* Kafka is said to have a better scalability and good support from the development community.
+
+=== Architecture Overview
+The project consists of multiple components:
+
+* Kafka Client Library - new command transport implementation, producer and consumer client logic.
+* Remoting Kafka Plugin - plugin implementation with KafkaGlobalConfiguration, KafkaComputerLauncher and KafkaSecretManager.
+* Remoting Kafka Agent - A custom JAR agent with remoting JAR packaged together with a custom Engine implementation to setup a communication channel with Kafka. The agent is also packaged as a https://hub.docker.com/r/jenkins/remoting-kafka-agent/[Docker image] in DockerHub.
+* All the components are packaged together with Docker Compose.
+
+The below diagram is the overview of the current architecture:
+image:/images/post-images/remoting-kafka/remoting-kafka-architecture.png[title="Remoting Kafka Plugin Current Architecture", role="center"]
+
+With this design, master is not communicating with agent using direct TCP communication anymore, all the communication commands are transfered with Kafka.
+
+=== Phase 1 Summary
+
+* Setup project as a set of Docker Compose components: Kafka cluster, Jenkins master (with plugin) and a custom agent.
+* Create a PoC with command transport implementation to support Kafka, which involves of command invocation,RMI, classloading and data streaming.
+* Make necessary changes in Remoting and Core to make them extensible (if needed).
+* Make a decision whether Kafka is suitable as a final implementation.
+
+==== Phase 1 Presentation Slides
```

Comment:
I doubt it makes sense to embed phase 1/2 slides and videos. I would just add links and embed only the final presentation